psl/psl-core/src/validate/datasource_loader.rs

```diff
@@ -49,85 +53,67 @@ fn lift_datasource(
     diagnostics: &mut Diagnostics,
     connectors: crate::ConnectorRegistry,
 ) -> Option<Datasource> {
-    let source_name = &ast_source.name.name;
-    let mut args: HashMap<_, _> = ast_source
+    let source_name = ast_source.name.name.as_str();
```

Comment:
This whole function starts to be too long. Not now, but maybe in a later PR we could cut it into smaller pieces?

Reply:
As I said, a morning refactor. A few hundred lines and a fast approval.